pubsub.go

```diff
@@ -0,0 +1,125 @@
+package lazyhttp
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/dendhi31/lazyhttp/redismaint"
+)
+
+func (httprequest *Client) Consumer() {
+	config := redismaint.Configuration{
+		RedisURL:   httprequest.PubSubServer,
+		ContexName: "first",
+		Debug:      true,
+		Handler:    httprequest.SendRequestWithPubSub,
+	}
+
+	rmaint, err := redismaint.New(config)
+	if err != nil {
+		log.Fatalln(err.Error())
+		return
+	}
+
+	term := make(chan os.Signal, 1)
+	signal.Notify(term, os.Interrupt, syscall.SIGTERM)
+
+	go func() {
+		log.Println("starting the lazyhttp consumer")
+		rmaint.Run()
+	}()
+	//send sample schedule
+	select {
+	case <-rmaint.Err():
+		log.Println("err", err)
+	case <-term:
+		rmaint.Stop()
+		log.Println("signal terminated detected")
+	}
+}
+
+func (httprequest *Client) SendRequestWithPubSub(ctx context.Context, url string, action string, payload []byte, header map[string]string, key string) (int, []byte, error) {
```

Comment:
@dendhi31 
I think we should not tell the user how the request is handle, this is too confusing.

This is just my 2 cents: 
1. instead of function `SendRequest` we can use a terminology like `Get` or `Post` i.e `client.Get(....)` or `client.Post(....)`
2. The process behind it should be configure on initialization, i.e: use_cache or not. Thus we can use strategic pattern. 
3. If !useCache, the logic should use pessimistic method, otherwise, we can use redis + pubsub.